File: Day-06-of-30-Days/Solution.js
// Day 06 of 30 Days 👾

function binaryToDecimal(binary) {
    if(binary == "") {
        return "Invalid Binary Number";
    }

    let dec = 0;
    for(let i=0; i<binary.length; i++) {
        if(binary[i] == '1' || binary[i] == '0') {
            dec = dec * 2 + parseInt(binary[i]);
        } else {
            throw new Error("Invalid Binary Number");
        }
    }
    return dec;
}

// example 1
console.log(binaryToDecimal("1011")); // Answer: 11

// example 2
console.log(binaryToDecimal("111")); // Answer: 7

File: Day-06-of-30-Days/Solution.py
# Day 06 of 30 Days 👾

def binary_to_decimal(binary):
    if binary == "":
        print("Invalid binary number")
        return 0
    decimal = 0
    for bin in binary:
        if bin == '1' or bin == '0':
            decimal = decimal * 2 + int(bin)
        else:
            raise ValueError("Invalid Binary Number")
    return decimal


# example 1
print(binary_to_decimal("1100")) # ans: 12

# example 2
print(binary_to_decimal("1000")) # ans: 8